cmd2/argcomplete_bridge.py

```diff
@@ -0,0 +1,248 @@
+                     default_completer=argcomplete.FilesCompleter()):
+            """
+            :param argument_parser: The argument parser to autocomplete on
+            :type argument_parser: :class:`argparse.ArgumentParser`
+            :param always_complete_options:
+                Controls the autocompletion of option strings if an option string opening character (normally ``-``) has not
+                been entered. If ``True`` (default), both short (``-x``) and long (``--x``) option strings will be
+                suggested. If ``False``, no option strings will be suggested. If ``long``, long options and short options
+                with no long variant will be suggested. If ``short``, short options and long options with no short variant
+                will be suggested.
+            :type always_complete_options: boolean or string
+            :param exit_method:
+                Method used to stop the program after printing completions. Defaults to :meth:`os._exit`. If you want to
+                perform a normal exit that calls exit handlers, use :meth:`sys.exit`.
+            :type exit_method: callable
+            :param exclude: List of strings representing options to be omitted from autocompletion
+            :type exclude: iterable
+            :param validator:
+                Function to filter all completions through before returning (called with two string arguments, completion
+                and prefix; return value is evaluated as a boolean)
+            :type validator: callable
+            :param print_suppressed:
+                Whether or not to autocomplete options that have the ``help=argparse.SUPPRESS`` keyword argument set.
+            :type print_suppressed: boolean
+            :param append_space:
+                Whether to append a space to unique matches. The default is ``True``.
+            :type append_space: boolean
+
+            .. note::
+                If you are not subclassing CompletionFinder to override its behaviors,
+                use ``argcomplete.autocomplete()`` directly. It has the same signature as this method.
+
+            Produces tab completions for ``argument_parser``. See module docs for more info.
+
+            Argcomplete only executes actions if their class is known not to have side effects. Custom action classes can be
+            added to argcomplete.safe_actions, if their values are wanted in the ``parsed_args`` completer argument, or
+            their execution is otherwise desirable.
+            """
+            self.__init__(argument_parser, always_complete_options=always_complete_options, exclude=exclude,
+                          validator=validator, print_suppressed=print_suppressed, append_space=append_space,
+                          default_completer=default_completer)
+
+            if "_ARGCOMPLETE" not in os.environ:
+                # not an argument completion invocation
+                return
+
+            global debug_stream
```

Comment:
Global variable **debug_stream** isn't defined at the module level.  Should it be?

Also, is this really a constant?  If so maybe use all uppercase and put it at the module level?

Reply:
I probably need to import that from argcomplete.